waku/waku_store_sync/storage/storage.nim

```diff
@@ -0,0 +1,33 @@
+import std/math, results, chronos
+
+import ../../waku_core/time, ../common
+
+type SyncStorage* = ref object of RootObj
+
+method insert*(
+    self: SyncStorage, element: ID
+): Result[void, string] {.base, gcsafe, raises: [].} =
+  discard
```

Comment:
Maybe `insert` should be idempotent?

Reply:
Currently if you can insert with the same element the first call would be `ok()` but if you call again `err()` since the element was already present.

An idempotent function would always insert the same element at the same index and would result in no side effect when calling the same func with the same params multiple times.